model/RedeemOffer.js

```
@@ -0,0 +1,49 @@
+const low = require('lowdb')
+const FileSync = require('lowdb/adapters/FileSync')
+const adapter = new FileSync('db.json')
+const db = low(adapter)
+
+module.exports = {
+    RedeemOffer: function RedeemOffer(offersAvailable, accountnumber){
+
+        db.read()
+        let offerData = []
+        offerData = db.get('offers').value()
+
+        let redeemOfferResultList = []
+        let transactionIdCount = db.get('transactionId').value()
+
+        var j = 0
```

I see where you are getting with this loop with the j variable. It makes sense to me but we also want to check if the offer couldn't be found in our offer pool. If that was the case we would return an error in the response status. To do that, you would have to rewrite your loop. 

I recommend using foreach function, foreach allows you to do something per item in an array. If you look at my RedeemComp.js or any of those, you can see how I use that. With that it's easier to read, and we can return an error message if we can't find the offer.